blockchain/src/history/mmr_store.rs

```diff
@@ -143,29 +159,33 @@ fn key_to_index(key: Vec<u8>) -> Option<(u32, usize)> {
 
 impl<'a, 'env> Store<Blake2bHash> for MMRStore<'a, 'env> {
     fn push(&mut self, elem: Blake2bHash) {
-        if let Tx::Write(ref mut tx) = self.tx {
+        // This function assumes that there is no higher epoch.
+        // Otherwise the append method will fail.
+        if let Some(ref mut cursor) = self.cursor {
             let key = index_to_key(self.epoch_number, self.size);
-            tx.put(self.hist_tree_table, &key, &elem);
+            cursor.append(&key, &elem);
             self.size += 1;
         }
```

Comment:
This should panic otherwise, as you must not push an element on a read-only store.